DXMainClient/DXGUI/Multiplayer/GameInformationPanel.cs

```diff
@@ -156,15 +216,80 @@ public void ClearInfo()
             lblHost.Visible = false;
             lblPing.Visible = false;
             lblPlayers.Visible = false;
+            lblGameInformation.Visible = false;
 
             foreach (XNALabel label in lblPlayerNames)
                 label.Visible = false;
+
+            if (mapTexture != null && !mapTexture.IsDisposed)
+                mapTexture.Dispose();
+
+            if (pingTexture != null && !pingTexture.IsDisposed)
+                pingTexture.Dispose();
         }
 
         public override void Draw(GameTime gameTime)
         {
             if (Alpha > 0.0f)
+            {
                 base.Draw(gameTime);
+
+                if (game != null)
+                {
+                    if (pingTexture != null)
+                        DrawTexture(pingTexture, new Rectangle(lblPing.ClientRectangle.X + 26, lblPing.Y, pingTexture.Width, pingTexture.Height), Color.White);
+
+                    if (mapTexture != null)
+                        RenderMapPreview();
+                }
+            }
+        }
+
+        private void RenderMapPreview()
+        {
+            int maxPreviewHeight = 150;
+
+            // Calculate map preview area based on right half of ClientRectangle
+            double xRatio = (ClientRectangle.Width / 2 - 10) / (double)mapTexture.Width;
+            double yRatio = (ClientRectangle.Height - 20) / (double)mapTexture.Height;
+
+            double ratio = Math.Min(xRatio, yRatio); // Choose the smaller ratio for scaling
+            int textureWidth = (int)(mapTexture.Width * ratio);
+            int textureHeight = (int)(mapTexture.Height * ratio);
+
+            // Apply max height constraint
+            if (textureHeight > maxPreviewHeight)
+            {
+                ratio = maxPreviewHeight / (double)mapTexture.Height;
+                textureHeight = maxPreviewHeight;
+                textureWidth = (int)(mapTexture.Width * ratio); // Recalculate width to maintain aspect ratio
+            }
+
+            int texturePositionX = rightColumnPositionX + (ClientRectangle.Width / 2 - textureWidth) / 2; // Center in the right column
+            int texturePositionY = mapPreviewPositionY;
+
+            DrawTexture(
+                mapTexture,
+                new Rectangle(texturePositionX, texturePositionY, textureWidth, textureHeight),
+                Color.White
+            );
+        }
+
+        private Texture2D GetTextureForPing(int ping)
+        {
+            switch (ping)
+            {
+                case int p when (p > 350):
+                    return pingTextures[4];
+                case int p when (p > 250):
+                    return pingTextures[3];
+                case int p when (p > 100):
+                    return pingTextures[2];
+                case int p when (p >= 0):
+                    return pingTextures[1];
+                default:
+                    return pingTextures[0];
+            }
```

Comment:
Pretty sure this is duplicated code, unify it with game lobby one somehow (and ping texture loading too I think)

Reply:
It is, where do I unify this in one place?